azurelinuxagent/common/protocol/goal_state.py

```diff
@@ -135,19 +131,41 @@ def fetch_goal_state(wire_client):
         """
         return GoalState(wire_client)
 
+    @staticmethod
+    def _verify_gs_for_noop(goal_state):
+        noop_gs_msg_format = "[PERIODIC] Goal State [incarnation {inc}] retrieved, but its a no-op."
+
+        if not goal_state.ext_conf:
+            logger.periodic_warn(logger.EVERY_SIX_HOURS, noop_gs_msg_format.format(inc=goal_state.incarnation))
+            raise ValueError("Goal State is a no-op")
+
+
     @staticmethod
     def fetch_full_goal_state(wire_client):
         """
         Fetches the full goal state, including nested properties (such as extension config).
         """
-        return GoalState(wire_client, full_goal_state=True)
+        goal_state = GoalState(wire_client, full_goal_state=True)
+
+        try:
+            GoalState._verify_gs_for_noop(goal_state)
+        except ValueError:
+            return None
```

Comment:
we shouldn't return None... the caller needs to handle the missing role instance. (e.g. the logic to refresh the host ga plugin should probably retry; the extension handler should skip extension processing, etc)

Reply:
similar comment below (fetch_full_goal_state_if_incarnation_different_than)

Reply:
I think the precedent for these methods was to return `None`. e.g. 

> the extension handler should skip extension processing

already happens with the `None` returned. Would the simplest answer be to add the host ga refresh retry logic based on a `is None` check?